### PR TITLE
refactor(text)!: fully remove spacing and text passage variants

### PR DESCRIPTION
**Pull request by jinlee93**
[EDS-837]
### Summary:
- with preflight, can now remove the `spacing` and `as="div"` text passage variants
- replaced the spacing provided with TW utility classes
  - text passage variant used `<LayoutLinelengthContainer>` which just applied max width of 36 rem, replaced with className `max-w-xl`
### Test Plan:
- compare the affected stories with the ones from https://chanzuckerberg.github.io/edu-design-system/
  - some small spacing differences are acceptable

### CODE REVIEW

**Comment by booc0mtaco** [comment]
@jinlee93 is merging this one blocked by #1490 ?

**Comment by jinlee93** [comment]
> @jinlee93 is merging this one blocked by #1490 ?

Yup, the PR target is that feature branch, so once that one lands, this one will auto-target `next`

**Comment by booc0mtaco** [comment]
> > @jinlee93 is merging this one blocked by #1490 ?
> 
> Yup, the PR target is that feature branch, so once that one lands, this one will auto-target `next`

Oh i see, thx

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx [hunk 1]
```diff
@@ -268,8 +268,8 @@ export const CoursePlannerEdit = () => {
   const container1EmptyContent = () => {
     return (
       <>
-        <Text as="div" className="!mb-8">
-          <p>There are no more available projects to create your course plan</p>
+        <Text className="mb-8 max-w-xl">
+          There are no more available projects to create your course plan
```
Comment (booc0mtaco) [inline .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:272]: As part of the Cookbook RFC work, we will review such stories and make the text and such more generic (and use that FPO block)

File: src/components/FieldNote/FieldNote.stories.tsx [hunk 2]
```diff
@@ -26,16 +26,16 @@ export const Default: StoryObj<Args> = {
...
-        <ul>
-          <li>Multiple lines</li>
-          <li>Arbitrary HTML text</li>
-          <li>
+      <div className="max-w-xl">
+        <Text className="mb-6">Here is a field note that involves:</Text>
+        <ul className="ml-4 list-disc">
+          <li className="mb-0">Multiple lines</li>
+          <li className="mb-0">Arbitrary HTML text</li>
+          <li className="mb-0">
```
Comment (booc0mtaco) [inline src/components/FieldNote/FieldNote.stories.tsx:34]: I would have thought preflight would make resetting the bottom margin unnecessary? Unless this is doing something else...

File: src/components/Text/Text.module.css [hunk 3]
```diff
@@ -74,173 +74,3 @@
...
-  margin-bottom: var(--eds-size-half);
-}
-
-.text--1x-spacing {
-  margin-bottom: var(--eds-size-1);
-}
-
-.text--2x-spacing {
-  margin-bottom: var(--eds-size-2);
-}
```
[the remark this answers is not in the document]
Comment (jinlee93) [reply src/components/Text/Text.module.css:91]: Totally meant to remove that prop entirely, fixed

File: src/components/FieldNote/FieldNote.stories.tsx (hunk 2 above)
Comment (jinlee93) [reply src/components/FieldNote/FieldNote.stories.tsx:34]: Yup, these classes are redundant, removed

File: .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx (hunk 1 above)
Comment (jinlee93) [reply .storybook/pages/CoursePlannerEdit/CoursePlannerEdit.tsx:272]: Super down for that

File: src/components/Text/Text.module.css [hunk 4]
```diff
@@ -74,173 +74,3 @@
...
-   */
-  blockquote {
-    font-style: italic;
-    border-left-width: var(--eds-theme-border-width);
-    border-left-style: solid;
-    border-left-color: var(--eds-theme-color-border-brand-primary-strong);
-    padding-left: var(--eds-size-2);
-    margin-left: 0;
-    margin-bottom: var(--eds-size-3);
-  }
```
[the remark this answers is not in the document]
Comment (jinlee93) [reply src/components/Text/Text.module.css:217]: Yup, doesn't need to be part of this text passage variant haha

File: src/components/Text/Text.tsx [hunk 5]
```diff
@@ -97,16 +88,12 @@ export const Text = forwardRef(
         'Info variant is deprecated, please consider another variant.',
       );
     }
-    const TagName =
-      capLinelength && as === 'div' ? LayoutLinelengthContainer : as;
+    const TagName = as;
```
[the remark this answers is not in the document]
Comment (jinlee93) [reply src/components/Text/Text.tsx:91]: What would that look like?

[the remark this answers is not in the document]
Comment (booc0mtaco) [reply src/components/Text/Text.tsx:85]: [Similar to this in `ClickableStyle`](https://github.com/chanzuckerberg/edu-design-system/blob/main/src/components/ClickableStyle/ClickableStyle.tsx#L103). Just a little syntax sugar.

[the remark this answers is not in the document]
Comment (jinlee93) [reply src/components/Text/Text.tsx]: Got it, changed

**Review by booc0mtaco — APPROVED** [review]
will review the combined ticket once this one is merged in to the preflight stuff

**Pull request merged**